wolfcrypt/src/asn.c

```diff
@@ -5612,8 +5642,8 @@ static int DecodePolicyOID(char *out, word32 outSz, byte *in, word32 inSz)
             return ASN_PARSE_E;
         }
 
-        /* Validate total length (2 is the CERT_POLICY_OID+SEQ) */
-        if ((total_length + 2) != sz) {
+        /* Validate total length */
+        if (total_length > (sz - (int)idx)) {
```

Comment:
The sz was 1 greater than (total_length + 2), so the != sz check was failing. I believe this was broken due to the ASN refactor. See 2805 second-cert-in-chain.pem for example.